subprojects/packagefiles/libssh2/meson.build

```diff
@@ -0,0 +1,196 @@
+libssh2_config_private_data.set('HAVE_STDLIB_H', libssh2_have_stdlib_h)
+libssh2_config_private_data.set('HAVE_SYS_SELECT_H', libssh2_have_sys_select_h)
+libssh2_config_private_data.set('HAVE_SYS_UIO_H', libssh2_have_sys_uio_h)
+libssh2_config_private_data.set('HAVE_SYS_SOCKET_H', libssh2_have_sys_socket_h)
+libssh2_config_private_data.set('HAVE_SYS_IOCTL_H', libssh2_have_sys_ioctl_h)
+libssh2_config_private_data.set('HAVE_SYS_TIME_H', libssh2_have_sys_time_h)
+libssh2_config_private_data.set('HAVE_SYS_UN_H', libssh2_have_sys_un_h)
+libssh2_config_private_data.set('HAVE_WINDOWS_H', libssh2_have_windows_h)
+libssh2_config_private_data.set('HAVE_WS2TCPIP_H', libssh2_have_ws2tcpip_h)
+libssh2_config_private_data.set('HAVE_WINSOCK2_H', libssh2_have_winsock2_h)
+libssh2_config_private_data.set('HAVE_NTDEF_H', libssh2_have_ntdef_h)
+libssh2_config_private_data.set('HAVE_NTSTATUS_H', libssh2_have_ntstatus_h)
+libssh2_config_private_data.set('HAVE_O_NONBLOCK', libssh2_have_o_nonblock)
+libssh2_config_private_data.set('HAVE_FIONBIO', libssh2_have_fionbio)
+libssh2_config_private_data.set('HAVE_EVP_AES_128_CTR', libssh2_have_EVP_aes_128_ctr)
+libssh2_config_private_data.set('HAVE_GETTIMEOFDAY', libssh2_have_gettimeofday)
+libssh2_config_private_data.set('HAVE_INET_ADDR', libssh2_have_gettimeofday)
+libssh2_config_private_data.set('HAVE_POLL', libssh2_have_poll)
+libssh2_config_private_data.set('HAVE_SELECT', libssh2_have_select)
+libssh2_config_private_data.set('HAVE_SOCKET', libssh2_have_socket)
+libssh2_config_private_data.set('HAVE_STRTOLL', libssh2_have_strtoll)
+libssh2_config_private_data.set('HAVE_STRTOI64', libssh2_have_strtoi64)
+libssh2_config_private_data.set('HAVE_SNPRINTF', libssh2_have_snprintf)
+libssh2_config_private_data.set('HAVE_LONGLONG', 1)
+libssh2_config_private_data.set('LIBSSH2DEBUG', get_option('buildtype') == 'debug')
+
+libssh2_config_private_h = configure_file(input: 'src/libssh2_config_cmake.h.in',
+    output: 'libssh2_config.h',
+    configuration: libssh2_config_private_data,
+    format: 'cmake',
+    install: true,
+    install_dir: 'include/'
+)
+libssh2_sources += [libssh2_config_private_h]
+
+# libssh2_config_cmake.h.in has no HAVE_NETINET_IN_H, HAVE_ARPA_INET_H
+if libssh2_have_netinet_in_h
+  add_project_arguments(['-DHAVE_NETINET_IN_H'], language: 'c')
+endif
+if libssh2_have_arpa_inet_h
+  add_project_arguments(['-DHAVE_ARPA_INET_H'], language: 'c')
+endif
+
+inc_dir = include_directories(['include', '.'])
+libssh2_lib = library('ssh2',
+    libssh2_sources,
+    include_directories: inc_dir,
+    dependencies: libssh2_deps,
+    c_args: extra_flags,
+    install: true
+)
+libssh2_dep = declare_dependency(
+    include_directories: inc_dir,
+    link_with: libssh2_lib
+)
+
+install_headers(
+    files('include/libssh2.h',
+          'include/libssh2_publickey.h',
+          'include/libssh2_sftp.h'
+    )
+)
+
+pkgconfig.generate(libssh2_lib)
+
+if get_option('examples').enabled()
```

Comment:
Why is this a feature option? You only use it to see if it is enabled, so if it is "auto" you treat it like "false".

Instead of a tristate option, why not just use a boolean option?